resources/metadata/abapEnvironmentRunATCCheck.yaml

```diff
@@ -0,0 +1,104 @@
+metadata:
+  name: abapEnvironmentRunATCCheck
+  description: Runs an ATC Check
+  longDescription: |
+    Run ATC Check
+spec:
+  inputs:
+    secrets:
+      - name: credentialsId
```

Comment:
please use credential naming as described here:
https://github.com/SAP/jenkins-library/blob/master/.github/CONTRIBUTING.md#jenkins-credential-handling

Reply:
Aliases for credentials seem to work now.
I would propose to use
`secrets:
   - name: cfCredentialsId
      aliases:
         - name: abapCredentialsId
`

Reply:
I made the respective changes for abapEnvironmentPullGitRepo with PR https://github.com/SAP/jenkins-library/pull/1520

Reply:
Open question: if both cfCredentialsId and abapCredentialsId is configured in the GENERAL section of the config.yml, which one takes precedence?

Reply:
From looking at the [code](https://github.com/SAP/jenkins-library/blob/master/pkg/config/config.go#L77), aliases are iterated in the order that they have been defined, but the code is only executed, if the parameter value is still `nil`. So in the above situation, `abapCredentialsId` would be used.

However, as Oliver said, the direct name of the secret should be prefixed with the tool. So maybe:

```
secrets:
   - name: abapCredentialsId
      aliases:
         - name: cfCredentialsId
```

?

Reply:
Indeed. I tested above situation. In that case, `abapCredentialsId` will be used. So: the first mentioned alias with value will be used.
For this step it would be very misleading to name the secret `abapCredentialsId`. @DanielMieg and I would prefer this implementation:
```
secrets:
   - name: credentialsId
      aliases:
         - name: abapCredentialsId
         - name: cfCredentialsId
```

Reply:
After discussing with @OliverNocon and @DanielMieg, changing the secret name to `abapEnvironmentCredentialsId` would mean an incompatible change for `abapEnvironmentPullGitRepo` when keeping it consistently. The following suggestion would mean an imcompatible change:
```
secrets:
   - name: abapEnvironmentCredentialsId
      aliases:
         - name: abapCredentialsId
         - name: cfCredentialsId
```
To keep it consistent and not to make incompatible changes I would still prefer to use the following implementation:
```
secrets:
   - name: credentialsId
      aliases:
         - name: abapCredentialsId
         - name: cfCredentialsId
```

Reply:
Another suggestion:
we use the suggestion from @stippi2 for abapEnvironmentRunATCCheck
´´´
secrets:
   - name: abapCredentialsId
      aliases:
         - name: cfCredentialsId
´´´
and for abapEnvironmentPullGitRepo
´´´
secrets:
   - name: abapCredentialsId
      aliases:
         - name: cfCredentialsId
         - name: credentialsId
´´´
to remain compatible

Reply:
@DanielMieg and I agreed on using above suggestion. Has been changed accordingly and tested.